test/date_time_parser_test.exs

```diff
@@ -185,6 +185,27 @@ defmodule DateTimeParserTest do
     )
 
     test_datetime_parsing("0000000000", DateTime.from_naive!(~N[1970-01-01T00:00:00], "Etc/UTC"))
+
+    test_datetime_parsing("-0000000001", DateTime.from_naive!(~N[1969-12-31T23:59:59], "Etc/UTC"))
+
+    test_datetime_parsing(
+      "-0000000001.0000000001",
+      DateTime.from_naive!(~N[1969-12-31T23:59:58.000000], "Etc/UTC")
+    )
+
+    # example from the Wikipedia article
+    test_datetime_parsing("-0386380800", DateTime.from_naive!(~N[1957-10-04T00:00:00], "Etc/UTC"))
+    test_datetime_parsing("-9999999999", DateTime.from_naive!(~N[1653-02-10T06:13:21], "Etc/UTC"))
+    # this matches Ruby
```

Comment:
It seems that when you go to 11 digits you go wrap around again and this is handled differently in different places.

From Ruby:

```ruby
Time.at(-99999999999).utc
=> -1199-02-15 14:13:21 UTC
```

From [epoch converter](https://www.epochconverter.com/):

![Screen Shot 2019-12-29 at 2 57 58 PM](https://user-images.githubusercontent.com/691365/71562724-719cae80-2a52-11ea-8102-12ccb9a2d4ae.png)

Since Elixir seemed to do what Ruby did, I included it and added a test with a comment.

Reply:
i think epochconverter.com is assuming the number is in milliseconds, hence the difference. I don't think we care about that here yet.